script.js

```
@@ -0,0 +1,454 @@
+        // Castling will be handled here later as a special type of king move.
+
+        return false;
+    }
+
+    // --- Turn Management ---
+
+    // NEW FUNCTIONS FOR CHECK DETECTION:
+
+    function findKing(kingColor) {
+        const kingPieceCode = kingColor.charAt(0) + 'K';
+        for (let r = 0; r < 8; r++) {
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)